src/NuGet.Clients/NuGet.PackageManagement.UI/Models/NuGetProjectUpgradeDependencyItem.cs

```diff
@@ -5,18 +5,51 @@
 using NuGet.Packaging.Core;
 using System.Globalization;
 using System.Linq;
+using NuGet.Common;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
 
 namespace NuGet.PackageManagement.UI
 {
-    public class NuGetProjectUpgradeDependencyItem
+    public class NuGetProjectUpgradeDependencyItem : INotifyPropertyChanged
     {
+        private bool _promoteToTopLevel;
         public PackageIdentity Package { get; }
         public IList<PackageIdentity> DependingPackages { get; }
 
+        public IList<PackLogMessage> Issues { get; }
+
+        public string Id { get; }
+
+        public string Version { get; }
+
+        public bool PromoteToTopLevel {
+            get
+            {
+                return _promoteToTopLevel;
+            }
+            set
+            {
+                _promoteToTopLevel = value;
+                OnPropertyChanged("PromoteToTopLevel");
+            }
+        }
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        protected void OnPropertyChanged([CallerMemberName] string propertyName = null)
+        {
+            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+        }
+
         public NuGetProjectUpgradeDependencyItem(PackageIdentity package, IList<PackageIdentity> dependingPackages = null)
         {
+            _promoteToTopLevel = true;
```

Comment:
Shouldn't this default to false, and only set to true for dependencies which are actually promoted?

Reply:
it sets it correctly on first pass through packages. The default view in UI will be false for transitive dependencies.